slave/slave_channels.go

```
@@ -0,0 +1,43 @@
+package slave
+
+import (
+	"fmt"
+
+	"github.com/aaronang/cong-the-ripper/lib"
+)
+
+func addTask(task lib.Task, s *Slave) {
+	taskStatus := lib.TaskStatus{
+		Id:       task.ID,
+		JobId:    task.JobID,
+		Status:   lib.Running,
+		Progress: task.Start,
+	}
+	s.heartbeat.TaskStatus = append(s.heartbeat.TaskStatus, taskStatus)
+}
+
+func password_found(Id int, password string, s *Slave) {
+	fmt.Println("Found password: " + password)
+	ts := taskStatusWithId(Id, s)
+	if ts != nil {
```

`ts` should always be found right? we should at least log an error instead of failing silently.

---

Agree, will log error if taskStatus is not found. Should indeed not happen.